// sms-service/lib/index.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SMSService = void 0;
const aws_sdk_1 = require("aws-sdk");
const debug = require("debug");
class SMSService {
    constructor(awsRegion = 'us-east-1') {
        this.snsClient = new aws_sdk_1.SNS({ region: awsRegion });
        this.logger = debug('sms-service');
    }
    sendSMS(phoneNumber, messageBody) {
        return __awaiter(this, void 0, void 0, function* () {
            const requestParams = {
                Message: messageBody,
                PhoneNumber: phoneNumber,
            };
            try {
                const response = yield this.snsClient.publish(requestParams).promise();
                this.logger('publish response: ' + JSON.stringify(response));
                return 'Success';
            }
            catch (err) {
                console.error(err, err.stack);
                throw new Error('Error: exception caught while attempting to send message');
            }
        });
    }
}
exports.SMSService = SMSService;
